lib/pericope/parsing.rb

```diff
@@ -109,30 +110,48 @@ def parse_ranges(book, ranges)
         range_begin.chapter = to_valid_chapter(book, range_begin.chapter)
         range_begin.verse = to_valid_verse(book, range_begin.chapter, range_begin.verse)
 
-        range_end = parse_reference_fragment(range_end_string, default_chapter: (range_begin.chapter unless chapter_range))
-        range_end.chapter = to_valid_chapter(book, range_end.chapter)
-
-        # treat Mark 3-1 as Mark 3-3 and, eventually, Mark 3:1-35
-        range_end.chapter = range_begin.chapter if range_end.chapter < range_begin.chapter
-
-        # this is a range of chapters, end with the last verse
-        if range_end.needs_verse?
-          range_end.verse = get_max_verse(book, range_end.chapter)
+        if range_begin_string == range_end_string && !chapter_range
+          range_end = range_begin.dup
         else
-          range_end.verse = to_valid_verse(book, range_end.chapter, range_end.verse)
+          range_end = parse_reference_fragment(range_end_string, default_chapter: (range_begin.chapter unless chapter_range))
+          range_end.chapter = to_valid_chapter(book, range_end.chapter)
+
+          # treat Mark 3-1 as Mark 3-3 and, eventually, Mark 3:1-35
+          range_end.chapter = range_begin.chapter if range_end.chapter < range_begin.chapter
+
+          # this is a range of chapters, end with the last verse
+          if range_end.needs_verse?
+            range_end.verse = get_max_verse(book, range_end.chapter)
+          else
+            range_end.verse = to_valid_verse(book, range_end.chapter, range_end.verse)
+          end
         end
 
         # e.g. parsing 11 in 12:1-8,11 => remember that 12 is the chapter
         default_chapter = range_end.chapter
 
-        Range.new(range_begin.to_verse(book: book), range_end.to_verse(book: book))
+        # e.g. parsing c in 9:12a, c => remember that 12 is the verse
+        default_verse = range_end.verse
+
+        range = Range.new(range_begin.to_verse(book: book), range_end.to_verse(book: book))
+
+        # an 'a' at the beginning of a range is redundant
+        range.begin.letter = nil if range.begin.letter == "a" && range.end.to_i > range.begin.to_i
+
+        # a 'c' at the end of a range is redundant
```

Comment:
⬇️⬇️ You set `max_letter` to "d" above. Guessing that should match in the comment?